app/editor/src/features/admin/products/ProductDetailsForm.tsx

```diff
@@ -0,0 +1,180 @@
+import { useFormikContext } from 'formik';
+import { debounce, noop } from 'lodash';
+import moment from 'moment';
+import React from 'react';
+import { useApp } from 'store/hooks';
+import { useNotifications, useReports, useUsers } from 'store/hooks/admin';
+import {
+  Col,
+  FieldSize,
+  FormikCheckbox,
+  FormikDatePicker,
+  FormikSelect,
+  FormikText,
+  FormikTextArea,
+  getEnumStringOptions,
+  getUserOptions,
+  IOptionItem,
+  IProductModel,
+  OptionItem,
+  ProductTypeName,
+  Row,
+  Show,
+} from 'tno-core';
+
+/**
+ * The page used to view and edit reports.
+ * @returns Component.
+ */
+export const ProductDetailsForm: React.FC = () => {
+  const { values, setFieldValue } = useFormikContext<IProductModel>();
+  const [{ userInfo }] = useApp();
+  const [{ users }, { findUsers }] = useUsers();
+
+  const [{}, apiNotifications] = useNotifications();
+  const [{}, apiReports] = useReports();
+  const [targetProductOptions, setTargetProductOptions] = React.useState<IOptionItem[]>([]);
+
+  const [userOptions, setUserOptions] = React.useState(getUserOptions(users.items));
+
+  const productTypeOptions = getEnumStringOptions(ProductTypeName);
+
+  React.useEffect(() => {
+    if (userInfo?.id) {
+      findUsers({ quantity: 50, includeUserId: values.ownerId })
+        .then((results) => {
+          setUserOptions(getUserOptions(results.items));
+        })
+        .catch(() => {});
+    }
+    // Only fire on initial load.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userInfo?.id]);
+
+  React.useEffect(() => {
+    if (values?.productType && values?.targetProductId >= 0) {
+      changeTargetProduct(values.productType);
+    }
+  }, [values?.productType, values?.targetProductId]);
+
+  const handleFindUsers = debounce(async (text: string) => {
+    const results = await findUsers({ quantity: 50, username: text }, true);
+    setUserOptions(getUserOptions(results.items));
+    return results;
+  }, 500);
+
+  const changeTargetProduct = React.useCallback(
+    (targetProduct: ProductTypeName | undefined) => {
+      if (!targetProduct) return;
+      switch (targetProduct) {
+        case ProductTypeName.Report:
+          // set using reports
+          apiReports.findAllReports().then((data) => {
+            setTargetProductOptions(data.map((s) => new OptionItem(s.name, s.id)));
+          });
+          break;
+        case ProductTypeName.EveningOverview:
+          // set list to empty
+          setTargetProductOptions([]);
+          setFieldValue('targetProductId', -1);
+          break;
+        case ProductTypeName.Notification:
+          // set using notifications
+          apiNotifications.findAllNotifications().then((data) => {
+            setTargetProductOptions(data.map((s) => new OptionItem(s.name, s.id)));
+          });
+          break;
+      }
+    },
+    [targetProductOptions, apiReports, apiNotifications],
```

Comment:
If you are intentionally not including all dependencies place a comment explaining why